src/main/java/org/myrobotlab/service/MouthControl.java

```diff
@@ -24,26 +28,69 @@
 
   private static final long serialVersionUID = 1L;
   public final static Logger log = LoggerFactory.getLogger(MouthControl.class);
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int mouthClosedPos = 20;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int mouthOpenedPos = 4;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int delaytime = 75;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int delaytimestop = 150;
+  @Deprecated /* future releases members will be protected use appropriate methods instead */
   public int delaytimeletter = 45;
+  
   transient private ServoControl jaw;
   transient private SpeechSynthesis mouth;
+  protected String jawName;
+  protected String mouthName;
 
   public MouthControl(String n, String id) {
     super(n, id);
+    registerForInterfaceChange(ServoControl.class);
+    registerForInterfaceChange(SpeechSynthesis.class);
   }
-
+  
   public void attach(Attachable attachable) {
     if (attachable instanceof SpeechSynthesis) {
       mouth = (SpeechSynthesis) attachable;
-      mouth.attachSpeechListener(this);
+      subscribe(mouth.getName(), "publishStartSpeaking");

```

Comment:
why not just use the attachSpeechListener(this) call?  we're duplicating code and implementation this way..